src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,44 @@
+package core.basesyntax;
+
+import static core.basesyntax.Color.WHITE;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    public static final int LENGTH_LIMIT = 250;
+
+    public Figure getRandomFigure() {
+        Figure newFigure = null;
+        Color color = ColorSuplier.getRandomColor();
+        int figureIndex = new Random().nextInt(FigureTypes.values().length);
+        FigureTypes newFigureType = FigureTypes.values()[figureIndex];
+        switch (newFigureType) {
+            case CIRCLE: int radius = new Random().nextInt(LENGTH_LIMIT);
+                newFigure = new Circle(color, radius);
+                break;
+            case SQUARE: int side = new Random().nextInt(LENGTH_LIMIT);
+                newFigure = new Square(color, side);
+                break;
+            case RECTANGLE: int side1 = new Random().nextInt(LENGTH_LIMIT);
+                int side2 = new Random().nextInt(LENGTH_LIMIT);
+                newFigure = new Rectangle(color, side1, side2);
+                break;
+            case RIGRHTTRIANGLE: int leg1 = new Random().nextInt(LENGTH_LIMIT);
+                int leg2 = new Random().nextInt(LENGTH_LIMIT);
+                newFigure = new RightTriangle(color, leg1, leg2);
+                break;
+            case ISOSCELESTRAPEZOID: int height = new Random().nextInt(LENGTH_LIMIT);
+                int lowerBase = new Random().nextInt(LENGTH_LIMIT);
+                int upperBase = new Random().nextInt(LENGTH_LIMIT);
```

Comment:
Let's create only one object of a type Random and then reuse it
```suggestion
            case RECTANGLE: int side1 = random.nextInt(LENGTH_LIMIT);
                int side2 = random.nextInt(LENGTH_LIMIT);
                newFigure = new Rectangle(color, side1, side2);
                break;
            case RIGRHTTRIANGLE: int leg1 = random.nextInt(LENGTH_LIMIT);
                int leg2 = random.nextInt(LENGTH_LIMIT);
                newFigure = new RightTriangle(color, leg1, leg2);
                break;
            case ISOSCELESTRAPEZOID: int height = random.nextInt(LENGTH_LIMIT);
                int lowerBase = random.nextInt(LENGTH_LIMIT);
                int upperBase = random.nextInt(LENGTH_LIMIT);
```

Reply:
Also, you can increase the readability of this method by moving figure creating to separate private methods.
Remember, your methods shouldn't be more than 15-20 lines long